integrations/huggingface-transformers/tutorials/sparse-transfer-learning-bert.md

```diff
@@ -0,0 +1,343 @@
+```bash
+zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none
+```
+
+SparseML uses the stub to download the model before starting training.
+
+### Kick off Training
+
+Run the following to fine-tune the pre-sparsified BERT onto SST2:
+
+```bash
+sparseml.transformers.text_classification \
+  --output_dir pruned_quantized_obert-text_classification_sst2 \
+  --model_name_or_path zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none \
+  --recipe zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/pruned90_quant-none \
+  --distill_teacher zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/base-none \
+  --task_name sst2 --max_seq_length 128 --per_device_train_batch_size 32 --per_device_eval_batch_size 32 --preprocessing_num_workers 6 \
+  --do_train --do_eval --evaluation_strategy epoch --fp16  \
+  --save_strategy epoch --save_total_limit 1
+```
+
+Lets discuss the key arguments:
+
+- `--task` specifies the dataset to train on (in this case SST2). You can pass any GLUE task to the `--task` command. Check out the use case pages for details on passing a custom dataset.
+
+- `--model_name_or_path` specifies the starting checkpoint for the training process. Here, we passed a SparseZoo stub, which
+identifies the 90% pruned BERT model in the SparseZoo. The script downloads the PyTorch model to begin training. In addition to SparseZoo stubs, you can also pass a local path to a PyTorch checkpoint.
+
+- `--recipe` specifies the transfer learning recipe. Recipes are YAML files that declare the sparsity related algorithms
+ that SparseML should apply. For transfer learning, the recipe instructs SparseML to maintain sparsity during training
+ and to apply quantization over the final epochs. In this case, we passed a SparseZoo stub, which instructs SparseML
+ to download a premade SST2 transfer learning recipe. In addition to SparseZoo stubs, you can also pass a local path to a YAML recipe. 
+ See below for more details on what the transfer learning recipe looks like.
+
+- `--distill_teacher` is an optional argument that allows you to apply model distillation during fine-tuning. Here, we pass SparseZoo stub (ending in `base_none`, specifying the dense version) which pulls a dense BERT model trained on SST2 from the SparseZoo.
+
+The SparseML script downloads the model, teacher, and dataset and recipe. Recipes are critical to the SparseML system, as they declare the algorithms and hyperparameters applied by SparseML. In this case, the premade transfer learning recipe for SST2 controls the number of epochs, the learning rate, as well as instructions for how to apply pruning, quantization, and distillation. SparseML parses the recipe and updates the training loop with the specified algorithms and hyperparameters encoded therein before starting training.
```

Comment:
The SparseML script downloads the model, teacher, ~~and~~ dataset and recipe

Reply:
In this case, the premade transfer learning recipe for SST2 controls the number of epochs, learning rate, and instructions for applying pruning, quantization, and distillation.